### ISSUE TITLE
2.0: Bug with numbers (like 12345678)  storing in Swift?

### ISSUE THREAD

**Issue by vellori** [couchbase/couchbase-lite-ios#1871]
Hello,
I create a document this way (swift):
```
        let doc = Document(docName)
        let num = 12345678
        doc.set(num, forKey: "cid")
        do {
            try db.save(doc)
            print("Save Success")
        } catch let error {
            print(error.localizedDescription)
        }
```

I've found that if value of "num" is greater than 8388607, then the following return a negative value, like it overflows:
```
        let doc = db.getDocument(docName)
        print(doc?.int(forKey: "cid") ?? "NO ID")
        print(doc?.float(forKey: "cid") ?? "NO ID")
        print(doc?.double(forKey: "cid") ?? "NO ID")
        let num = doc?.double(forKey: "cid")
        let numInt = Int(num!)
        print("Int conversion \(numInt)")
```
Printout:
```
-4431538
-4.43154e+06
-4431538.0
Int conversion -4431538
```
I've looked thru docs to understand what data size I should expect but I wasn't to find it... either it's a Number size limitation for Ints or a bug. 

Interestingly, if you try using 123456789... printout is correct:
```
123456789
1.23457e+08
123456789.0
Int conversion 123456789
```

Maybe I'm doing something terribly wrong here?
--------
* Version:   2.0DB013
* Client OS: iOS 10
* Server:    No server

**Comment by pasin** [comment]
This seems to be a bug.

**Comment by snej** [comment]
This is a bug in the way certain integers are encoded; I've fixed it in Fleece and merged the change into LiteCore, but I can't merge the fix into Couchbase Lite yet because there are other new LiteCore changes (named indexes) that will break Couchbase Lite until it's updated to support them. (I think that's #1666?)

**Comment by pasin** [comment]
I'm almost done with #1666. I should be able to merge by EOD today.

**Comment by pasin** [comment]
The fix has been merged to [feature/2.0](https://github.com/couchbase/couchbase-lite-ios/tree/feature/2.0) branch. Thank you very much for reporting this.
